### ISSUE TITLE
macOS SecKeyChain

### ISSUE THREAD

**Issue by angelru** [dotnet/maui#7126]
### Description

This same code with a non-maui application retrieves the certificates installed on my mac, but in maui the SecStatusCode is ItemNotFound, am I missing something? i'm stuck at work right now.

### Steps to Reproduce

```
          SecRecord rec = new(SecKind.Certificate);
            SecStatusCode code;
            SecKeyChain.QueryAsRecord(rec, out code);
            NSData[] nsDatas = SecKeyChain.QueryAsData(rec, false, 999, out code);

            if (nsDatas != null)
            {
                foreach (NSData item in nsDatas)
                {

                    X509Certificate2 cert = new X509Certificate2(item.ToArray());
                    x509Certificate2s.Add(cert);
                }
            }
```

### Version with bug

Release Candidate 3 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

macOS

**Comment by jfversluis** [comment]
Is your non-MAUI app also a Mac Catalyst app? Maybe that's the difference?

**Comment by angelru** [comment]
> Is your non-MAUI app also a Mac Catalyst app? Maybe that's the difference?

It is a macOS Coca App that the above code works on, how can I make it work on Mac Catalyst app?

**Comment by jfversluis** [comment]
Looks like this is a restriction in Mac Catalyst: https://developer.apple.com/forums/thread/696431

> API Differences
>
> The SecItem API is well aligned with the data protection keychain. When you use it to talk to the file-based keychain the API has to work through a shim. That shim has limitations. Some of those limitations are inherent to the keychain implementation. For example, the access control model of the file-based keychain is very different from that of the data protection keychain, and the shim can’t make up for that. However, some limitations are just bugs. If you’re porting iOS keychain code to the Mac, it’s best to target the data protection keychain.
>
> Mac Catalyst apps only have access to the data protection keychain.
>
> iOS Apps on Mac only have access to the data protection keychain.

So if you can even get it to work, you'll have to find out how to do it with a Mac Catalyst app. This is not something that is because of a bug with us :)

**Comment by angelru** [comment]
@jfversluis is it possible to run a MAUI app for macOS without Mac Catalyst?

**Comment by jfversluis** [comment]
I don't think it would work with .NET MAUI. I think you should be able to add a macOS target, but you'll have to write the platform code for that as all of our .NET MAUI controls now leverage Mac Catalyst

**Comment by angelru** [comment]
Wow, I was hoping... I still can't find a way to access the certificates installed on a mac from Mac Catalyst. Thank you very much.
